go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -837,6 +844,37 @@ func (bsnc *BaseSecondaryNetworkController) buildUDNEgressSNAT(localPodSubnets [
 	return snats, nil
 }
 
+func (bsnc *BaseSecondaryNetworkController) ensureDHCP(pod *corev1.Pod, podAnnotation *util.PodAnnotation, lsp *nbdb.LogicalSwitchPort) error {
+	opts := []kubevirt.DHCPConfigsOpt{}
+
+	ipv4DNSServer, ipv6DNSServer, err := kubevirt.RetrieveDNSServiceClusterIPs(bsnc.watchFactory)
+	if err != nil {
+		return err
+	}
+
+	ipv4Gateway, _ := util.MatchFirstIPFamily(false /*ipv4*/, podAnnotation.Gateways)
+	if ipv4Gateway != nil {
+		opts = append(opts, kubevirt.WithIPv4Router(ipv4Gateway.String()))
+	}
+
+	if bsnc.MTU() > 0 {
+		opts = append(opts, kubevirt.WithIPv4MTU(bsnc.MTU()))
+	}
+
+	opts = append(opts, kubevirt.WithIPv4DNSServer(ipv4DNSServer), kubevirt.WithIPv6DNSServer(ipv6DNSServer))
+
+	return kubevirt.EnsureDHCPOptionsForLSP(bsnc.controllerName, bsnc.nbClient, pod, podAnnotation.IPs, lsp, opts...)
```

Comment:
ok I have 1 more stupid question. with regards to RAs in v6. I can see on the LSP we set ARPProxyIPv6's MAC which is a hardcoded value for VMs as the server_id. Same for v4. will the server_id be the same for L2 UDN pods and we don't care right cause its a dummy thing at this point since OVN is our DHCP server of sorts?

one more thing: for L3 live migration case we seem to set the ARPProxyIPv4 as the default gateway for the pod but in UDPN we will set the GR's LRP addr i.e .1 of pod subnet correct?

Reply:
> ok I have 1 more stupid question. with regards to RAs in v6. I can see on the LSP we set ARPProxyIPv6's MAC which is a hardcoded value for VMs as the server_id. Same for v4. will the server_id be the same for L2 UDN pods and we don't care right cause its a dummy thing at this point since OVN is our DHCP server of sorts?
> 

We would care only if there are multiple DHCP server, but I don't say users configuring VMs with primary UDN and also configuring a DHCP server.

> one more thing: for L3 live migration case we seem to set the ARPProxyIPv4 as the default gateway for the pod but in UDPN we will set the GR's LRP addr i.e .1 of pod subnet correct? - we hard code a value because we don't want migration to have gateway change?

Yep, that's exactly it, at L3 it was quite bad since not only MAC but gateway IP is different per node, and at ipv4 you cannot the path from a multipath default gw sending a message.